pkg/git/clone.go

```diff
@@ -0,0 +1,58 @@
+package git
+
+import (
+	"context"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/pkg/errors"
+)
+
+var defaultCloner repoCloner = &remoteRepoCloner{}
+
+// CloneRepo clones the repository in the given URL to the given dstPath and checks out the given revision.
+// revision can be 'main', a release version (e.g. 1.4.1), a commit hash (e.g. 34edf09a).
+func CloneRepo(url, dstPath, rev string) error {
+	repo, err := defaultCloner.Clone(url, dstPath, true)
+	if err != nil {
+		return errors.Wrapf(err, "Error downloading repository (%s)", url)
+	}
+	if rev != "" {
```

Comment:
I think setting rev to a default value like `main` or returning an error in case rev is empty would be better than swallowing the problem and just returning nil.